hack/lint-shell.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 T-Mobile, USA, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Trademark Disclaimer: Neither the name of T-Mobile, USA, Inc. nor the names of
+# its contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+
+################################################################################
+#### Variables, Arrays, and Hashes #############################################
+################################################################################
+
+RUN_TYPE="${1}"
+SELECTION_FILE="hack/.shellcheck-selection"
+
+################################################################################
+#### Main ######################################################################
+################################################################################
+
+# check to see if being run for ci, if yes exclude legacy scripts from linting
+if [[ "ci" == "${RUN_TYPE}" ]]; then
```

Comment:
Just a small nit here. It's usually preferred to compare the variable to the variable to the assumed value.

```suggestion
if [[ "${RUN_TYPE}" == "ci" ]]; then
```

Reply:
I do this because it avoids accidental assignment. Let's change all the other stuff to be like this instead.